benches/README.md

```diff
@@ -0,0 +1,53 @@
+# Fuel Core Benches
+This directory contains a set of benchmarks for Fuel Core. The benchmarks are
+used for calculating gas metering costs for the Fuel Core VM.
+
+## Running the benchmarks
+To run the benchmarks, you can use `cargo bench -p fuel-core-benches` command available.
+Alternatively you can use `cargo criterion -p fuel-core-benches` if you have it installed.
+For more information on using criterion see [the guide](https://bheisler.github.io/criterion.rs/book/).
+
+## Profiling a benchmark
+Sometimes it is useful to produce a flamegraph from a benchmark to verify
+you are measuring the correct things.
+To do this you can use the `cargo flamegraph` [command](https://github.com/flamegraph-rs/flamegraph).
+
+### Benchmarking a specific function
+First build the benchmark with debug symbols:
+`CARGO_PROFILE_RELEASE_DEBUG=true cargo-criterion -p fuel-core-benches --no-run`
+Then you need to find the actual test binary that was built:
+`ls target/release/deps/vm* -lat | head -1`
+Should give you something like:
+`target/release/deps/vm-a17190f2ca5e7169`
+Then you can run the benchmark with the `flamegraph` command (runs the bench for 10 seconds):
+`flamegraph target/release/deps/vm-a17190f2ca5e7169 --bench bench_name --profile-time 10`
```

Comment:
```suggestion
`flamegraph -- target/release/deps/vm-a17190f2ca5e7169 --bench bench_name --profile-time 10`
```

Reply:
Hmm this is not needed for me. Maybe you have a different version of flamegraph?

Reply:
strange, maybe it'd be better to use cargo commands to choose the target and bench since flamegraph supports criterion?

either way this is a minor thing that doesn't need to be addressed now.